app/controllers/analysis_controller.rb

```diff
@@ -0,0 +1,420 @@
+                10..19 => "Ages 10-19",
+                20..29 => "Ages 20-29",
+                30..39 => "Ages 30-39",
+                40..49 => "Ages 40-49",
+                50..59 => "Ages 50-59",
+                60..69 => "Ages 60-69",
+                70..79 => "Ages 70-79",
+                80..89 => "Ages 80-89",
+                90..99 => "Ages 90-99",
+                100..120 => "Ages 100 and up"}
+    all_ages = []
+    partial_age_hash = {}
+=begin For each of the ages specified, place them in the partial_age_hash hash
+    which will be used later to lookup the appropriate value when creating the key for decedent_frequency.
+=end
+    decedent_age_range.each do |val|
+      if val < 100
+        (val..val + 9).each do |age|
+          all_ages.push(age)
+          partial_age_hash[(val..val+9)] = age_keys[(val..val+9)]
+        end
+      else
+        (val..val + 20).each do |age|
+          all_ages.push(age)
+          partial_age_hash[(100..120)] = age_keys[(100..120)]
+        end
+      end
+    end
+    start_year = params[:start_year].to_i
+    end_year = params[:end_year].to_i
+    specified_cause_of_death = params[:cause_of_death]
+    decedent_residence_zip = params[:residence_zipcode]
+    decedent_death_zip = params[:death_zipcode]
+    num_causes = params[:num_causes]
+    criteria = JSON.parse(params[:specified_criteria])
+    decedent_race = JSON.parse(params[:race])
+    target_decedent_cods = []
+    # Array of all the required queries based on what is specified by the user.
+    params_specified = []
+    decedent_frequency = {}
+    if !decedent_race.empty?
+      decedent_race.each do |race|
+        decedent_frequency.keys.each {|k| decedent_frequency[race + " " + k] = decedent_frequency[k]; decedent_frequency.delete(k)}
+      end
+    end
+    initial_query = "contents ->> 'sex.sex' = 'Male' or contents ->> 'sex.sex' = 'Female'"
+    if decedent_sex.length == 1
+      initial_query = "contents ->> 'sex.sex' = '" + decedent_sex[0] +"'"
+    end
+    params_specified.push(initial_query)
+    if !decedent_residence_zip.empty?
+      params_specified.push("contents ->> 'decedentAddress.zip' = '" + decedent_residence_zip + "'")
+    end
+
+    if !decedent_death_zip.empty?
+      params_specified.push("contents ->> 'locationOfDeath.zip' = '" + decedent_death_zip + "'")
+    end
+
+    if criteria.include?("served_armed_forces")
+      params_specified.push("contents ->> 'armedForcesService.armedForcesService' = 'Yes'")
+    end
+
+    if criteria.include?("tobacco_cause")
+      params_specified.push("contents ->> 'didTobaccoUseContributeToDeath.didTobaccoUseContributeToDeath' = 'Yes'")
+    end
+
+    if num_causes.include?(".") || num_causes.include?("-")
+      render json: {}
+      return
+    else
+      num_causes = num_causes.to_i
+    end
+    # Run a separate query if both genders are requested for.
+    result_query = DeathRecord.where(initial_query)
+    result_query = DeathRecord
```

Comment:
This is confusing to me, you are setting `result_query` to the `DeathRecord`s filtered by `initial_query`, but then you are setting it equal to the class? And then you filter on more queries it looks like below. This might need to be refactored a bit.